src/test/kotlin/org/jmailen/gradle/kotlinter/functional/CustomTaskTest.kt

```diff
@@ -0,0 +1,188 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import groovy.util.GroovyTestCase.assertEquals
+import java.io.File
+import org.gradle.testkit.runner.TaskOutcome
+import org.intellij.lang.annotations.Language
+import org.jmailen.gradle.kotlinter.functional.utils.editorConfig
+import org.jmailen.gradle.kotlinter.functional.utils.kotlinClass
+import org.jmailen.gradle.kotlinter.functional.utils.resolve
+import org.jmailen.gradle.kotlinter.functional.utils.settingsFile
+import org.junit.Before
+import org.junit.Test
+
+class CustomTaskTest : WithGradleTest.Kotlin() {
+
+    lateinit var projectRoot: File
+
+    @Before
+    fun setUp() {
+        projectRoot = testProjectDir.root.apply {
+            resolve("settings.gradle") { writeText(settingsFile) }
+            resolve("build.gradle") {
+                @Language("groovy")
+                val buildScript = """
+                    plugins {
+                        id 'kotlin'
+                        id 'org.jmailen.kotlinter'
```

Comment:
Here's one more line I would like to discuss. When I was performing code changes initially I wanted to read values from the extension, but then I realized that one of plugin capabilities is to create standalone tasks: https://github.com/jeremymailen/kotlinter-gradle#custom-tasks
so the plugin itself doesn't have to be applied. Kotlinter dependency just needs to be added to project conifuration.

```suggestion
```

Therefore I started wondering if I should remove the plugin application from tests.

Reply:
I wouldn't worry too much about that case.